### DISCUSSION TITLE
API Reference for Microsoft.Extensions.AI

### DISCUSSION THREAD

**Discussion by BrandonStudio** [dotnet/ai-samples#95] [Q&A]
It is rather hard to work without documentation.
And examples in https://github.com/dotnet/ai-samples are outdated

**Comment by jmatthiesen** [comment]
Hi, @BrandonStudio - for now, you can find API documentation for Microsoft.Extensions.AI in the following location:

https://github.com/dotnet/extensions/tree/main/src/Libraries/Microsoft.Extensions.AI.Abstractions

We're also working to get these published in the official Microsoft Learn documentation, but its not quite there yet. In the meantime, feel free to ask questions here as well.

As for outdated samples - we're working to update this (starting with quick starts), but also have some Microsoft.Extensions.AI samples in the repo already. Please let me know which ones you see are outdated and would like to have us update!

**Reply by BrandonStudio** [reply]
Yes, I see changes in ec1000f :)

**Comment by schaveyt** [comment]
please please. can these docs be also published in a manner that is AI consumable. For example, 

- <https://docs.anthropic.com/llms-full.txt>
- <https://directory.llmstxt.cloud/>

**Reply by jmatthiesen** [reply]
Oh this is super interesting - I hadn't seen this yet, thanks for sharing!
